src/Lumper.Lib/Jobs/StripperJob.Block.cs

```diff
@@ -18,43 +19,36 @@ protected abstract partial class Block
         protected static Prop ParseProp(string line, int lineNr)
         {
             Match match = PairRegex().Match(line);
-            if (match.Success)
-            {
-                var pair = new Prop(
-                    match.Groups[1].Value,
-                    match.Groups[2].Value);
-                return pair;
-            }
-            else
-            {
-                throw new NotImplementedException($"Can't parse KeyValuePair '{line}' in line {lineNr}");
-            }
+            if (!match.Success)
+                throw new InvalidDataException($"Can't parse KeyValuePair '{line}' in line {lineNr}");
+
+            return new Prop(match.Groups[1].Value, match.Groups[2].Value);
         }
-        protected static void ParseBlock(StreamReader reader, bool blockOpen, ref int lineNr, Action<string, int> fun)
+
+        protected static void ParseBlock(StreamReader reader, bool blockOpen, ref int lineNr, Action<string, int> fn)
         {
-            string line;
-            while ((line = reader.ReadLine()) != null)
+            while (reader.ReadLine() is { } line)
```

Comment:
It's pattern matching syntax that says "while reader.ReadLine() returns a non-null object, assign to `line`)".

Thinking about it though, this could just be `reader.ReadLine is string line`. Rider complains with some Resharper warning but who cares, `is string line` is way more readable than `is { } line`

Reply:
Actually, eh. `is string` makes it seems like it's narrowing on some union type (though I guess C# doesn't even have those), not performing a null check. Whilst `is { } line` is quite weird at first, if you're familiar with pattern matching syntax it's the most explicit way to show the point of the pattern is to do a null check.